client/src/pages/Prospects.js

```diff
@@ -60,44 +61,169 @@ const useStyles = makeStyles((theme) => ({
 
 const Prospects = () => {
   const classes = useStyles();
+  const actionType = 'Add to Campaign'
 
-  const prepareData = () => {
-    const results = [];
-
-    //replace data with real data
-    const data = SampleData();
-    const cloudIcon = <CloudIcon className="fas fa-cloud" style={{color: "grey"}} />
-    data.map(each => {
-      const obj = {
-        'check': 'check',
-        'Email': each.email,
-        cloudIcon,
-        'Status': 'working',
-        'Owner': each.owner,
-        'Campaigns': each.campaigns,
-        'Last Contacted': each.lastContacted,
-        'Emails...': each.emails
+  const [data, handleData] = useState([{}]);
+  const [dialog, handleDialog] = useState(null);
+  const [listOfCampaigns, handleCampaigns] = useState(null);
+  const [campaignId, setCampaignId] = useState('');
+  const [selectedProspects, handleSelectedProspects] = useState([]);
+
+  useEffect(() => {
+    getAllCampaigns();
+    getAllProspects();
+  }, [])
+
+  const getAllProspects = () => {
+    fetch(`/prospects`, {
+      method: 'GET',
+      headers: {
+        'Authorization': `Bearer ${getJWT()}`
+      }
+    })
+    .then(res => res.json())
+      .then(result => {
+        const listOfProspects = [];
+        const cloudIcon = <CloudIcon className="fas fa-cloud" style={{color: "grey"}} />
+
+        result.Prospects.map(prospect => {
+          const prospectObj = {
+            'id': prospect.id,
+            'check': 'check',
+            'Email': prospect.email,
+            cloudIcon,
+            'Status': 'working',
+            'Owner': prospect.name,
+            'Campaigns': prospect.campaigns,
+            'Last Contacted': prospect.lastContacted,
+            'Emails...': prospect.emails
+          }
+          return listOfProspects.push(prospectObj)
+        })
+        handleData(listOfProspects)
+      })
+    .catch(err => {
+      console.log(err.message);
+    });
+  }
+
+  const getAllCampaigns = () => {
+    fetch(`/campaigns`, {
+      method: 'GET',
+      headers: {
+        'Authorization': `Bearer ${getJWT()}`
       }
-      return results.push(obj)
     })
-    return results;
+    .then(res => res.json())
+      .then(data => {
+        handleCampaigns(data.Campaigns)
+    })
+    .catch(err => {
+      console.log(err.message);
+    });
+  }
+
+  const handleCloseDialogAndSaveProspects = () => {
+    handleDialog(false)
+    saveProspectsToCampaign()
+  } 
+
+  const saveProspectsToCampaign = () => {
+    const data = {
+      "prospect_ids": selectedProspects,
+    };
+
+    fetch(`/campaign/${campaignId}/prospects`, {
+      method: "POST",
+      headers: {
+        'Authorization': `Bearer ${getJWT()}`,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
```

Comment:
Is this header necessary?

Reply:
Do you mean the Accept header? If yes, it is not needed. I will be removing it on my next push